src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -66,8 +68,8 @@ internal partial class DiagnosticIncrementalAnalyzer : IIncrementalAnalyzer2
 
         internal DiagnosticAnalyzerInfoCache DiagnosticAnalyzerInfoCache => _diagnosticAnalyzerRunner.AnalyzerInfoCache;
 
-        public bool IsCompilationEndAnalyzer(DiagnosticAnalyzer diagnosticAnalyzer, Project project, Compilation compilation)
-            => DiagnosticAnalyzerInfoCache.IsCompilationEndAnalyzer(diagnosticAnalyzer, project, compilation) == true;
+        public async Task<bool> IsCompilationEndAnalyzerAsync(DiagnosticAnalyzer diagnosticAnalyzer, Project project, CancellationToken cancellationToken)
+            => await DiagnosticAnalyzerInfoCache.IsCompilationEndAnalyzerAsync(diagnosticAnalyzer, project, cancellationToken).ConfigureAwait(false) == true;
```

Comment:
`== true`?  seems odd.

Reply:
Yes, this is existing issue with how to deal with analyzer throwing an exception when we execute analyzer callbacks to determine `IsCompilationEndAnalyzer`. I am going to try to get rid of all this logic in favor of an actual API for analyzers to delcare themselves as compilation wide analyzers in a follow-up PR.